subworkflows/local/coverage_stats.nf

```diff
@@ -29,13 +29,14 @@ workflow COVERAGE_STATS {
     ch_bam = SAMTOOLS_VIEW.out.bam.join(ch_csi)
     
     // Calculate Coverage 
-    MOSDEPTH(ch_bam, ch_bed, fasta.map{it -> it[1]})
+    MOSDEPTH(ch_bam, ch_bed.map{it -> it[1]}, fasta.map{it -> it[1]})
     ch_versions = ch_versions.mix(MOSDEPTH.out.versions)
 
     emit:
     global = MOSDEPTH.out.global_txt
     summary = MOSDEPTH.out.summary_txt
-    bed = MOSDEPTH.out.per_base_bed
-    base = MOSDEPTH.out.per_base_csi
+    base_bed = MOSDEPTH.out.per_base_bed
+    base_csi = MOSDEPTH.out.per_base_csi
+    bed = ch_bed
```

Comment:
I guess the rename is motivated by the arrival of this new `ch_bed` channel ? Since the latter contains the fasta_windows regions, I would call it `fasta_windows_bed` (or `fw_bed` to make it short). That would remove the case for renaming the other two channels. If you still want to rename them, use the full name, i.e. `per_base_bed/csi`.

But wait a sec. Are you sure it's the `per_base` output of mosdepth that we need ? It's the `regions` files that have the windows defined by fasta_windows.

Reply:
I've added the `regions.bed` for the `coverage` module. I can't remember why base_bed/csi were in there originally but I can remove them.